pkg/monitor/monitor.go

```diff
@@ -204,14 +172,6 @@ func (m *Monitor) waitForServiceDescriptorDeletion(ctx context.Context, name str
 }
 
 func (m *Monitor) logFailureDetails(ctx context.Context, retErr error) {
-	var serviceInstance zap.Field
```

Comment:
we need to log these details

Reply:
it should be good enough to log full SD and LD as LD contains detailed failure reasons for individual resources

Reply:
Yeah, the problem is we don't know what ServiceInstances exist for a given ServiceDescriptor. By removing this code path we just always log the full SD and any errors associated with it.

Reply:
1. Of course we do know. If reporter is able to figure out SIs we are able to do it too.
2. We don't need to because all details are in LD anyways.

Reply:
I mean there might be some obscure cases when Voyager is so broken than LD has fake news in it, but 🤷‍♂️

Reply:
@benbarclay as I pointed above, we could just log reporter output, but it's going to be HUGE. Adding LD is probably good enough actually, as it contains bubbled up errors from ServiceInstances etc (but SD doesn't!)

Reply:
I'll add LD for now, but that just seems like a hack around the problems with error propagation to SD

Reply:
It's not a hack, it's by design. See the relevant doc. You were on that meeting btw 😴

Reply:
Dumping reporter output is preferable to reaching for other objects.

Reply:
According to the spec, the SD status is supposed to have the status of each location. If we're dropping the correct status of a location such that we need to pull the LD to get that information, then it's an issue and this is a hack.

Reply:
I tend to agree with Ben. Even if at first we will only show a cluster-local status (i.e. it won't include status of LD from the secondary cluster), it will be a huge improvement already.

Reply:
SD contains status of each location, but no resource statuses. Thus it is useless.